.github/workflows/pull-request-build-connector-template.yml

```diff
@@ -0,0 +1,51 @@
+name: PR Build (Connector)
+
+on:
+  workflow_call:
+    inputs:
+      additional-build-flags:
+        required: false
+        type: string
+        default: ""
+      additional-test-flags:
+        required: false
+        type: string
+        default: ""
+
+
+jobs:
+    ubuntu-build:
+        name: Build on Ubuntu
+        runs-on: ubuntu-latest
+        concurrency:
+          group: ${{ github.head_ref }}-ubuntu-build
+          cancel-in-progress: true
+        steps:
+            -   name: Checkout Repository
+                uses: actions/checkout@v1
+
+            - name: Set Up Ballerina
+              uses: ballerina-platform/setup-ballerina@v1.1.0
+              with:
+                version: latest
+
+            -   name: Set up JDK 17
+                uses: actions/setup-java@v3
+                with:
+                    distribution: 'temurin'
+                    java-version: 17.0.7
+            
+            -   name: Set ENV Variables
+                run: |
+                  echo -e '${{ toJson(secrets) }}' | jq -r 'to_entries[] | .key + "=" + .value' >> $GITHUB_ENV 
+
+            -   name: Build the Package
+                env:
+                    packageUser: ${{ github.actor }}
+                    packagePAT: ${{ secrets.GITHUB_TOKEN }}
+                run: |
+                  ./gradlew build -x test ${{ inputs.additional-build-flags }}
+                  ./gradlew test ${{ inputs.additional-test-flags }}
+
+            -   name: Generate Codecov Report
+                uses: codecov/codecov-action@v2
```

Comment:
I think we need to use codecov-action v3 as well. Please check and update.

Reply:
Updated with [f91e8c9](https://github.com/ballerina-platform/ballerina-standard-library/pull/4838/commits/f91e8c976848391ac77970102c5a830d415094e8)